app/pages/adduser/contextCreator.test.js

```diff
@@ -1,33 +1,82 @@
 import manifest from './manifest.json';
-import { getContext } from './contextCreator';
+import { getContext, getErrorContext } from './contextCreator';
+import { errorMessages } from './errorStrings';
 
 const mockData = {
   organisationId: 'org1',
   name: 'Manchester',
 };
 
-describe('getContext', () => {
-  it('should return the contents of the manfest', () => {
-    const context = getContext({});
-    expect(context.title).toEqual(manifest.title);
-    expect(context.description).toEqual(manifest.description);
-    expect(context.orgNameSubheading).toEqual(manifest.orgNameSubheading);
-    expect(context.questions.length).toEqual(manifest.questions.length);
-    context.questions.forEach((question, i) => {
-      expect(question.id).toEqual(manifest.questions[i].id);
-      expect(question.text).toEqual(manifest.questions[i].text);
+const mockValidationErrors = [{
+  field: 'firstName',
+  id: 'FirstNameRequired',
+},
+{
+  field: 'firstName',
+  id: 'FirstNameTooLong',
+}];
+
+describe('adduser contextCreator', () => {
+  describe('getContext', () => {
+    it('should return the contents of the manifest', () => {
+      const context = getContext({});
+      expect(context.title).toEqual(manifest.title);
+      expect(context.description).toEqual(manifest.description);
+      expect(context.orgNameSubheading).toEqual(manifest.orgNameSubheading);
+      expect(context.questions.length).toEqual(manifest.questions.length);
+      context.questions.forEach((question, i) => {
+        expect(question.id).toEqual(manifest.questions[i].id);
+        expect(question.text).toEqual(manifest.questions[i].text);
+      });
+      expect(context.addUserButtonText).toEqual(manifest.addUserButtonText);
+    });
+
+    it('should add organisationId and organisationName from the data provided', () => {
+      const context = getContext(mockData);
+      expect(context.organisationId).toEqual(mockData.organisationId);
+      expect(context.organisationName).toEqual(mockData.name);
     });
-    expect(context.addUserButtonText).toEqual(manifest.addUserButtonText);
-  });
 
-  it('should add organisationId and organisationName from the data provided', () => {
-    const context = getContext(mockData);
-    expect(context.organisationId).toEqual(mockData.organisationId);
-    expect(context.organisationName).toEqual(mockData.name);
+    it('should construct backLinkHref from the data provided', () => {
+      const context = getContext(mockData);
+      expect(context.backLinkHref).toEqual(`/organisations/${mockData.organisationId}`);
+    });
   });
+  describe('getErrorContext', () => {
```

Comment:
an additional describe block for `when they are no errors`?

Reply:
when there are no errors this function does not get called - the `getContext` function is called